app/controllers/salt_controller.rb

```diff
@@ -0,0 +1,31 @@
+require "velum/salt"
+# HealthController holds methods for checking the readiness of the application.
+class SaltController < ApplicationController
+  skip_before_action :redirect_to_setup
+
+  def update
+    minions = assigned_with_status
+    minions.each do |minion|
+      if minion.update_status == Minion.statuses[:update_needed]
```

Comment:
So move the `Minion.find_by(minion_id: minion.minion_id).update(highstate: Minion.highstates[:pending])` line to a method on the model ? Or the whole `assigned_with_status` as well?